### ISSUE TITLE
MD051: `§` survives the GitHub slug, and an emoji not surrounded by spaces gets one hyphen too many — both from the emoji marker pass

### ISSUE THREAD

**Issue by stasson** [rvben/rumdl#854]
### rumdl version

0.2.65. Reproduces identically on 0.2.35 and 0.2.63.

### Installation method

cargo install (also reproduced through `rumdl_lib::lint` as a library consumer)

### Bug description

The GitHub anchor style has an emoji marker pass (`process_emoji_sequences` in `src/utils/anchor_styles/github.rs`) that replaces each emoji with a `§EMOJI§` marker before the character pass, then rewrites the markers into hyphens at the end. It causes two divergences from GitHub.

**1. A section sign (`§`, U+00A7) in a heading survives the slug.** `§` is the marker's delimiter, and the character pass preserves it so the marker survives to the reassembly step. A `§` written in the heading is indistinguishable from one the pass inserted, so it survives too:

```rust
// src/utils/anchor_styles/github.rs:275 (unchanged from v0.2.35 to v0.2.65)
} else if c == '§' {
    // Preserve our marker character
    result.push(c);
}
```

GitHub strips the character, so for `## A §1 B` MD051 rejects `#a-1-b` (the anchor a browser resolves) and accepts `#a-§1-b` (an anchor no renderer produces). `§` is common in specification prose (`## 9. Worked Examples — the ridl §13 Contracts`), where every correct link to such a heading is reported.

**2. An emoji not surrounded by spaces gets one hyphen too many.** GitHub's anchor filter keeps `[\p{Word}\- ]`, drops every other character and turns each space into a hyphen, so an emoji is deleted and the spaces around it stay. The marker pass instead pops the spaces around the emoji and re-adds one or two hyphens by position, which is exact only when the emoji is surrounded by spaces:

| Heading | GitHub (github-slugger 2.0.0) | rumdl 0.2.65 |
| --- | --- | --- |
| `A 🚀 B` | `a--b` | `a--b` |
| `A🚀 B` | `a-b` | `a--b` |
| `A 🚀B` | `a-b` | `a--b` |
| `A🚀B` | `ab` | `a--b` |
| `A °🚀 B` | `a--b` | `a---b` |
| `Notes… 🚀 done` | `notes--done` | `notes---done` |
| `Notes… done` | `notes-done` | `notes-done` |

The last two rows show a related effect: the pass treats every `is_emoji_or_symbol` character as a symbol, a range that includes General Punctuation (`…`, `•`, `“`), but the pass only runs when the heading also contains an emoji from its quick-check ranges. So the slug of `…` depends on whether an emoji appears elsewhere in the same heading.

### Steps to reproduce

1. Create `repro.md`:

   ```markdown
   # T

   [l](#a-1-b)

   ## A §1 B

   x
   ```

2. Run `rumdl check repro.md`.

3. For the mirror, change the link to `[l](#a-§1-b)` and run again. This anchor resolves in no renderer.

4. For the emoji case, use `[l](#a-b)` and `## A🚀 B`.

### Expected behavior

`rumdl check repro.md` reports nothing: `#a-1-b` is the anchor GitHub produces for `## A §1 B`. The mirror, `[l](#a-§1-b)`, is reported. `#a-b` resolves `## A🚀 B`.

### Actual behavior

```console
$ rumdl check --no-config repro.md
repro.md:3:1: [MD051] Link anchor '#a-1-b' does not exist in document headings

Issues: Found 1 issues in 1 file (10ms)

$ rumdl check --no-config mirror.md

Success: No issues found in 1 file (7ms)
```

`heading_to_fragment("A §1 B")` returns `"a-§1-b"`; `heading_to_fragment("A🚀 B")` returns `"a--b"`.

### Configuration

None. Defaults, `flavor` unset.

### Operating system

macOS

### Additional context

**`§` is the only character that survives.** Substituting each of 73 characters GitHub strips (ASCII punctuation except `-` and `_`; Latin-1 U+00A1–U+00BF except the letters, numbers and soft hyphen; `×` `÷`; `– — ‘ ’ “ ” ‹ › † ‡ • … ‰ € ™ ≈ ≠ ≤ ≥ ∞`) into `## A X1 B` with the link left at `#a-1-b`, MD051 fires for exactly one of them: `§`. Every other symbol, including the others in the same Unicode category (`°`, `©`, `×`, `¶`), is stripped the way GitHub strips it. Same result on 0.2.35, 0.2.63 and 0.2.65.

**The pass is redundant with the character pass.** With `process_emoji_sequences` not called at all, every existing test in `github.rs` (33, including the emoji, flag, ZWJ-family and keycap cases) passes unchanged, and all the rows above match GitHub. ZWJ sequences, flags and skin-tone modifiers come out the same because each component is deleted and no space sits between them.

`kramdown_gfm.rs` uses the same delimiter for its own markers but is not affected: it preserves `§` only when it heads one of its marker strings, and a bare `§` falls through to the normal branch and is dropped.

A PR follows (linked below) that deletes the pass. The smaller alternative — keep the pass and drop the heading's own `§` before it runs — fixes the section sign only; it does not touch the hyphen count, since that comes from the pass itself. I have that variant ready as well if a one-line change is preferred for a first step.

Found through [prim](https://github.com/driftsys/prim), which runs MD051 as part of its selected rule set and carries a local workaround for the `§` case ([driftsys/prim#180](https://github.com/driftsys/prim/issues/180)) until it is fixed here.

**Comment by rvben** [comment]
Thanks @stasson, both claims were exactly right.

The GitHub anchor style replaced each emoji with a `§EMOJI§` marker before slugging, so a real `§` in a heading survived the pass, and an emoji not surrounded by spaces picked up an extra hyphen. Both are gone in v0.2.66: `## A §1 B` slugs to `a-1-b` and `## A🚀 B` to `a-b`, matching GitHub.
